libraries/SITL/SIM_Submarine.h

```diff
@@ -50,9 +50,24 @@ class Submarine : public Aircraft {
         float width  = 0.338; // y direction (meters)
         float height = 0.254; // z direction (meters)
         float weight = 10.5;  // (kg)
-        float net_bouyancy = 2.0; // (N)
+        float net_buoyancy = 2.0; // (N)
+
+        float buoyancy_acceleration = GRAVITY_MSS + net_buoyancy/weight;
+
+        // Frame drag coefficient
+        const Vector3f linear_drag_coefficient = Vector3f(0.2, 0.25, 0.3);
+        const Vector3f angular_drag_coefficient = Vector3f(0.25, 0.3, 0.2);
+        // Calculate total volume from water buoyancy
+        // $ V = F_b / (rho * g) $
+        // V = volume (m^3), rho = water density (kg/m^3), g = gravity (m/s^2), F_b = force (N)
+        float volume = buoyancy_acceleration * weight / (GRAVITY_MSS * 1023.6f);
+        // Calculate equivalent sphere area for drag force
+        // $ A = pi * r^2 / 4 $
+        // $ V = 4 * pi * r^3 / 3 $
+        // $ r ^2 = (V * 3 / 4) ^ (2/3) $
+        // A = area (m^3), r = sphere radius (m)
+        float equivalent_sphere_area = M_PI_4 * pow(volume * 3 / 4, 2 / 3);
```

Comment:
This is going to be *a* problem; you're doing integer arithmetic here, resulting in a lot of zeroes (and `pow(...)` returning 1...

Note sure if it is going to be *the* problem...